src/Models/CsvCatalogItem.js

```diff
@@ -59,20 +65,43 @@ var CsvCatalogItem = function(application, url) {
     this.data = undefined;
 
     /**
-     * Gets or sets the URL from which the {@link CsvCatalogItem#data} was obtained.
-     * @type {String}
+     * Gets or sets the tableStyle object
+     * @type {Object}
      */
-    this.dataSourceUrl = undefined;
+    this.tableStyle = undefined;
 
-    /**
-     * Gets or sets a value indicating whether data points in the CSV are color-coded based on the
-     * value column.
-     * @type {Boolean}
-     * @default true
+   /**
+     * Gets or sets the opacity (alpha) of the data item, where 0.0 is fully transparent and 1.0 is
+     * fully opaque.  This property is observable.
+     * @type {Number}
+     * @default 0.6
      */
-    this.colorByValue = true;
+    this.opacity = 0.6;
 
-    knockout.track(this, ['url', 'data', 'dataSourceUrl', 'colorByValue']);
+    knockout.track(this, ['url', 'data', 'tableStyle', 'opacity', '_clock']);
+
+    knockout.getObservable(this, 'opacity').subscribe(function(newValue) {
+        updateOpacity(this);
+    }, this);
+
+    delete this.__knockoutObservables.clock;
```

Comment:
I'll look at whether how it's being done in imageryLayer now can be used, but there were problems with the previous iteration of the clock in that.
________________________________________
From: Keith Grochow
Sent: Friday, 17 April 2015 11:02 AM
To: NICTA/nationalmap; NICTA/nationalmap
Subject: RE: [nationalmap] Region mapping updates (#635)

The time based datasets need to create their own clock settings that are used to update the main clock just like in imagery layer.  The actual creation of the clock is done in createRegionMappingClock which is called after the data is loaded and the setting can be determined.
________________________________________
From: Kevin Ring [notifications@github.com]
Sent: Thursday, 16 April 2015 3:24 PM
To: NICTA/nationalmap
Cc: Keith Grochow
Subject: Re: [nationalmap] Region mapping updates (#635)

In src/Models/CsvCatalogItem.js<https://github.com/NICTA/nationalmap/pull/635#discussion_r28484631>:

>
> -    knockout.track(this, ['url', 'data', 'dataSourceUrl', 'colorByValue']);
> +    knockout.track(this, ['url', 'data', 'tableStyle', 'opacity', '_clock']);
> +
> +    knockout.getObservable(this, 'opacity').subscribe(function(newValue) {
> +        updateOpacity(this);
> +    }, this);
> +
> +    delete this.__knockoutObservables.clock;


Maybe I'm missing something, but I don't see why the base class clock property needs to be replaced. Its replacement below looks functionally identical.

�
Reply to this email directly or view it on GitHub<https://github.com/NICTA/nationalmap/pull/635/files#r28484631>.

________________________________

The information in this e-mail may be confidential and subject to legal professional privilege and/or copyright. National ICT Australia Limited accepts no liability for any damage caused by this email or its attachments.